### ISSUE TITLE
Switching to Qt5

### ISSUE THREAD

**Issue by tresf** [LMMS/lmms#2611]
Many of our users are already compiling against Qt5, but most of them tend to be on Linux.  This topic is related to switching all builds to Qt5 permanently and what's needed prior to that switch.
- For starters, we need more testing.
  - Our first Win64 build which utilizes Qt5 is located here:<br> *(removed, please download 1.2.0 from the [downloads](https://lmms.io/download) page)*
  - Our first Mac build which utilizes Qt5 is located here:<br> *(removed, please download 1.2.0 from the [downloads](https://lmms.io/download) page)*
- Bugs (please check these off as they're fixed)
  - [x] All: Frequent crashes related to `QVector` https://github.com/LMMS/lmms/issues/2192#issuecomment-193905099.
  - [x] Mac: Spash window appears above SettingsDialog, MessageDialog (fixed via 1058ea4)
  - [x] All: Visual artifacts in MDI window decorations ([preview](https://cloud.githubusercontent.com/assets/6345473/13335913/6a2a6f48-dbe1-11e5-9497-720be1d11750.png)) (superseded by #2516)
  - [x] All: Menu bar colors are hard to read ([preview](https://cloud.githubusercontent.com/assets/6345473/13536660/2a22bb60-e20f-11e5-8b23-0fda60c21fa6.png))  (fixed via 5f48d1c)
  - [x] All: Inactive QMdiWindow titles are hard to read  ([preview](https://cloud.githubusercontent.com/assets/6345473/13536646/1a019436-e20f-11e5-81ec-56afdffe8a46.png)) (superseded by #2516)
  - [x] All: Language dropdown is blank (fixed via https://github.com/LMMS/lmms/commit/85011cdcf72fbba721bb0c10f140bfabf7a35725)
  - [x] All: White line on edge of all `QWidgets` ([preview](https://cloud.githubusercontent.com/assets/6345473/13538614/ab01bace-e219-11e5-8638-48204a41a6b7.png)) (superseded by #2516)
  - [X] All: `QMdiWindow` windowing controls are incorrectly colored ([preview](https://cloud.githubusercontent.com/assets/6345473/13336044/f96ca838-dbe1-11e5-8930-6a54b698b00a.png)) (`wont-fix`, see https://github.com/LMMS/lmms/issues/2611#issuecomment-193942180)
  - [x] All: Channel activity indicator [on by default](https://cloud.githubusercontent.com/assets/6345473/13509235/77c70a7c-e158-11e5-9701-b5734da7b756.png) (fixed via 8768769)
  - [x] All:  Drop-down menus are ["crammed"](https://cloud.githubusercontent.com/assets/6345473/13509753/1ce41188-e15b-11e5-8c68-94ebff14cdf3.png) (fixed via https://github.com/LMMS/lmms/commit/d098a39c76321293af73f398ab89965e827b4d12)
  - [x] All:  Colors incorrect for about dialog ([preview](https://cloud.githubusercontent.com/assets/6282045/13704648/3745f6f2-e79d-11e5-9f76-b3f6118079a0.png)) (fixed via #2781)
  - [x] All:  Crash when removing a channel that sends to another channel (#2667)
  - [x] Windows: Fix high-CPU usage for collapsed toolbars (#2767)
  - [x] All: VSTs GUI does not load properly (#2855) - Still open but will not affect Windows, Mac (fixed via #3786)
  - [x] Mac: ⌘ + drag shortcuts are broken (#2890) (old Xcode version only, not our bug)
  - [ ] Remove Mac/Windows Qt4 builds from Travis-CI

Screenshot, because everyone loves screenshots... 
![image](https://cloud.githubusercontent.com/assets/6345473/13336063/27578b78-dbe2-11e5-80ea-9356f4e7394a.png)

**Comment by liushuyu** [comment]
>Windows: Language dropdown is blank (see #1498 (comment))

Not only on Windows, but also on the other platforms namely Linux and Mac can't select languages

**Comment by tresf** [comment]
> In fact, I saw the binary translation files were generated, but not installed to the installation directory. But if you re-run make install, the files will appear in the correct directory. Strange.

@liushuyu thanks.  While looking for the language block you commented on, I noticed it was moved in https://github.com/LMMS/lmms/commit/ee3a99853b84d089fa99d4985d9d0b145d22bfb8.  Tagging @Lukas-W.

The language logic has moved quite a bit over the last year including moving from [`CMakeLists.txt`](https://github.com/LMMS/lmms/blob/df79601348ca3ed689f77ca128d3ff53f5802871/CMakeLists.txt#L545:L559) to [`src/CMakeLists.txt`](https://github.com/LMMS/lmms/blob/master/src/CMakeLists.txt#L140:L158) although I feel this may have a better home in [`data/locale/CMakeList.txt`](https://github.com/LMMS/lmms/blob/master/data/locale/CMakeLists.txt).

It's pretty obvious that order of operations has changed after this move, breaking language generation on first run for all platforms.

**Comment by Lukas-W** [comment]
> I feel this may have a better home in data/locale/CMakeList.txt.

Absolutely. Moved in 85011cdcf72fbba721bb0c10f140bfabf7a35725. That should fix the generation.

**Comment by tresf** [comment]
> Absolutely. Moved in 85011cd. That should fix the generation.

As it does. :+1: Checking it off here as I'll track the Windows/Travis stuff in #2577 since it's not Qt5 related.

**Comment by tresf** [comment]
Hmmm... I'm having no luck with the splash screen bug... Qt version is `5.5.1_2`

>  [  ] Mac: Spash window appears above SettingsDialog, MessageDialog

This bug doest **NOT** seem to occur on Windows.

I tried the following to no avail (clicking it won't raise it).

Solutions I tried:
 * https://forum.qt.io/topic/31197/unset-qt-windowstaysontophint-flag-problem/5

Yuck:
 * http://stackoverflow.com/a/17575185/3196753
 
No mention of it in known issues:
 * http://doc.qt.io/qt-5/osx-issues.html

Test results:
:no_entry: didn't work... (`GuiApplication.cpp`)
```diff
	// Show splash screen
	QSplashScreen splashScreen( embed::getIconPixmap( "splash" ) );
+	// Prevent splash from covering settings dialog
+	splashScreen.setWindowFlags( splashScreen.windowFlags() | Qt::WindowStaysOnBottomHint );
	splashScreen.show();
```

:no_entry: didn't work... (`GuiApplication.cpp`)
```diff
	// Show splash screen
	QSplashScreen splashScreen( embed::getIconPixmap( "splash" ) );
	splashScreen.show();
+	splashScreen.lower();
```

:no_entry: didn't work... (`SetupDialog.cpp`)
```diff
	vlayout->addWidget( settings );
	vlayout->addSpacing( 10 );
	vlayout->addWidget( buttons );
	vlayout->addSpacing( 10 );
	vlayout->addStretch();

+	setWindowFlags(windowFlags() | Qt::WindowStaysOnTopHint);

	show();

+	raise();
```

**Comment by softrabbit** [comment]
> Windows: Visual artifacts in MDI window decorations

I'm pretty sure I've seen those "ragged" MDI close buttons before, possibly in an LMMS/Qt5 build on Linux. Could they be a part of a style after all? And IMO they look too consistent for artifatcs.

**Comment by tresf** [comment]
> Could [["ragged" MDI close buttons](https://cloud.githubusercontent.com/assets/6345473/13335913/6a2a6f48-dbe1-11e5-9497-720be1d11750.png)] be a part of a style after all?

[Indeed they are](https://cloud.githubusercontent.com/assets/6345473/13504078/cece89d4-e140-11e5-98e4-893e4daaed43.pn).  @Umcaruje any interest in tracking this one down?

**Comment by Umcaruje** [comment]
> @Umcaruje any interest in tracking this one down?

My life is pretty crowded rn, I can take a look, but not sure when I'll have the time for it (I have some other issues I'm looking at).

Isn't #2516 going to fix this though? We are loading custom pixmaps for the buttons there.

**Comment by tresf** [comment]
> Isn't #2516 going to fix this though? We are loading custom pixmaps for the buttons there.

Well, it would certainly be a moot point if #2516 were to be implemented, but I'm not placing any bets on a hypothetical, and this would give us a better understanding of things, anyhow.

**Comment by Umcaruje** [comment]
>Well, it would certainly be a moot point if #2516 were to be implemented, but I'm not placing any bets on a hypothetical, and this would give us a better understanding of things, anyhow.

What if I started working on #2516 until its mergeable? ;) (though idk how that would be organised)

**Comment by tresf** [comment]
> What if I started working on #2516 until its mergeable? ;) (though idk how that would be organised)

Well, actually, that may be our only option... Here's Windows with the theme disabled... (still happens)

![image](https://cloud.githubusercontent.com/assets/6345473/13508861/ccc37be8-e156-11e5-8cf8-51029ff6a8db.png)

I'll take a second look at the code, but this may be a `not-our-bug`-type scenario after-all...

**Comment by tresf** [comment]
Adding new item to list...  

>  * [ ] Windows: Channel activity indicator [on by default](https://cloud.githubusercontent.com/assets/6345473/13509235/77c70a7c-e158-11e5-9701-b5734da7b756.png)

**Comment by tresf** [comment]
Update:

**Closed:**
 * Menu bar colors hard to read - fixed via 5f48d1c.
 * Mac splash screen hides SetupDialog - fixed via 1058ea4
 * Inactive QMdiWindow titles - superseded by #2516
 * Visual artifacts in MDI window decorations - superseded by #2516

**Critical/Open:**
 *  Channel activity indicator on by default (screenshot in description above)

**Cosmetic:**
  * MDI window controls are incorrectly colored (screenshot in description above)
  * Drop-down menus are "crammed" (screenshots below)

@Umcaruje can you tell me if this is the intended look of the dropdowns?  I ask because they seem a bit crammed and per #2567.  If they look fine, we'll remove it from the checklist.

| Mac | Windows |
|-------|-------------|
| ![image](https://cloud.githubusercontent.com/assets/6345473/13538230/55ec0be0-e217-11e5-9104-4398b6823117.png) | ![image](https://cloud.githubusercontent.com/assets/6345473/13538210/38681dd4-e217-11e5-8e77-2cc14a8e35e9.png)
 |

**Comment by Umcaruje** [comment]
Well, on linux, I actually have no icons in the dropdown menu, so I don't know how they looked.

I increased the padding accordingly in #2567, but I'm not sure how the icons behave. They should be moved 3 px to the left I guess, though Qt should render them centered, and not aligned to left.

If it can't be fixed in code, we can revert the margins back.

**Comment by Umcaruje** [comment]
Fixed it via https://github.com/LMMS/lmms/commit/d098a39c76321293af73f398ab89965e827b4d12

![screenshot from 2016-03-04 22 13 58](https://cloud.githubusercontent.com/assets/6282045/13540415/9a1b7294-e256-11e5-9a9a-07fb75228549.png)

**Comment by mikobuntu** [comment]
@Umcaruje Refering to your comment about no drop-down icons, this is a known issue... iirc I was talking to tobydox several years ago about this and if my memory server me right it was an issue with the actual DR.
 I wonder if there is a method to override this behaviour? Thanks

**Comment by Umcaruje** [comment]
@mikobuntu well, the problem is gone for me when I compiled using qt5, so I guess when we switch to it, it wouldn't matter.

**Comment by Umcaruje** [comment]
@tresf I rebased #2516 and the `White line on edge of all QWidgets` is also gonna be superseded by it: 
![screenshot from 2016-03-04 23 24 22](https://cloud.githubusercontent.com/assets/6282045/13542026/6573ee2c-e260-11e5-995c-1bcc5d98bec9.png)

**Comment by tresf** [comment]
> Fixed [menu icons] it via d098a39

:+1: 

> White line on edge of all QWidgets [...] superseded by #2516 

:+1: :+1: 

Do you guys get the LED indicator bug too on Linux?  I don't have a ready Qt5 Linux environment to test on.

**Comment by softrabbit** [comment]
> Do you guys get the LED indicator bug too on Linux? I don't have a ready Qt5 Linux environment to test on.

Yes, it's present.
LMMS 1.1.90-g2e2abdf
(Linux x86_64, Qt 5.4.2, GCC 5.2.1 20151010)

**Comment by tresf** [comment]
@softrabbit thanks for confirming.

Is there a chance this is just a bad initiation? Don't know the CSS side of things that well, but seems like this may be intended to be `Inactive` rather than `Active`.  Pinging @Umcaruje https://github.com/LMMS/lmms/blob/master/src/tracks/InstrumentTrack.cpp#L916:L920

**Comment by tresf** [comment]
Meh... nevermind that statement... that's more in regards to the QWidget being active or inactive from a user-interaction perspective.  I'm pretty sure Active is what we want, so this must be something else.

**Comment by tresf** [comment]
Ok, on second look, this is most likely due to the way Qt5 is handling the timer... 

https://github.com/LMMS/lmms/blob/master/src/gui/widgets/FadeButton.cpp#L86:L100

When the `FadeButton` is initially created, this statement is most likely coming back as true.  `m_stateTimer` is `QTime`.  According to the Qt5 docs, we should be using [`m_stateTimer.isNull()`](http://doc.qt.io/qt-5/qtime.html#isNull) as a check before doing a time comparison.  Qt4.8 states this function exists there as well. :+1:

**Comment by Lukas-W** [comment]
> we should be using m_stateTimer.isNull() as a check before doing a time comparison

Indeed, that fixes it. Fixed via 8768769450e245ef409fbe68cb6583c6253d90fc.

**Comment by tresf** [comment]
> Fixed via 

:+1: 

This next item is our last noted issue... 

>   [  ] All: `QMdiWindow` windowing controls are incorrectly colored ([preview](https://cloud.githubusercontent.com/assets/6345473/13336044/f96ca838-dbe1-11e5-8930-6a54b698b00a.png))

I've spent a considerable amount of time researching this to no avail.  If anyone knows how to style these little buggers, please chime in. :+1:

**Comment by liushuyu** [comment]
>This next item is our last noted issue...

Ah, I consider this as the most difficult issue among these... I suppose that this may caused by the different rendering/parsing behavior between Qt 4 and Qt 5.

**Comment by tresf** [comment]
@musikBear:  Here's your binary: https://github.com/tresf/lmms/releases/download/v1.2.0-RC1.2/lmms-1.1.90-win32-qt5.exe

Can we get some test results?  Some of the aforementioned bugs won't be patched in this release, but we need more test results before making the decision to switch.

**Comment by musikBear** [comment]
@tresf Thanks! :cake: I will install asa i get home, and test tonight. Sanity-tests, and are there anything else specific? Just let me know

**Comment by tresf** [comment]
> are there anything else specific

Yes, Qt5, specifically.  We need to know if we can switch to it as a permanent replacement for Qt4, so any major bugs that aren't in RC1's Qt4 version are precisely what we're looking for.

**Comment by musikBear** [comment]
@tresf 
Install was a little different. The install lib was asked for twice, and an error appears every time lmms opens from desktop-icon/ associated-link:
![qt5pr_drive_not_ready](https://cloud.githubusercontent.com/assets/6662066/13578872/b8f5b44a-e49a-11e5-934a-763644addff5.png)
A click on *continue* opens lmms normally.
This error does not appear, if a new or existing project is selected inside lmms.
**Graphics**
This is how the main windows look:
![qt5_all_mains](https://cloud.githubusercontent.com/assets/6662066/13578905/dc627dd2-e49a-11e5-9b75-4b6f162953c5.png)

Here are some instruments
![qt5_someinstruments](https://cloud.githubusercontent.com/assets/6662066/13578925/f4a37130-e49a-11e5-805f-68325214c093.png)

The sub-tabs:
![qt5_subpages](https://cloud.githubusercontent.com/assets/6662066/13578932/09537454-e49b-11e5-8609-2d3c00622e22.png)

One strange behavior is when mixer-window is moved:
![qt5_movingmixer](https://cloud.githubusercontent.com/assets/6662066/13578962/3425545e-e49b-11e5-838c-76454383c649.png)
The graphic 'hangs' / is stretched or some will disappear under the frame-border
![qt5_movingmixer2](https://cloud.githubusercontent.com/assets/6662066/13578995/5f85525c-e49b-11e5-98eb-b5b3241b41a8.png)
Unless you self, or other also see this odd mixer-blur, disregard as just my screen not updating. In empty projects, there is no blur!

Imo the windows-headers are in *current default-theme* difficult to read. *darkRebekka* would look much better, im sure :)

Besides graphics, is there something else that qt5 would influence in specific?

**Comment by tresf** [comment]
@musikBear 

Thanks for the detailed feedback.  Please continue to do so.

In regards to indicator lights and "x" button, both are documented in the original bug report as either fixed or superseded by a future fix.

The mixer artifacts are new... I'll try to reproduce.  Much obliged.

**Comment by tresf** [comment]
Also, Apple testing has been very minimal at this point, but we've received a small amount of feedback from @sunnystormy per https://github.com/LMMS/lmms/issues/2504#issuecomment-188522060.  The build @sunnystormy is using (linked in the original bug report) was built on OS X 10.11 "El Capitan" which won't work on anything older than Yosemite 10.10, so getting a Qt5 build that's crated with 10.8 tested has to happen before the permanent switch to Qt5 (I use 10.7 currently, but Qt5 isn't supported on 10.7).

**Comment by tresf** [comment]
@musikBear 

I'm unable to reproduce the Mixer graphical glitches https://www.youtube.com/watch?v=aMVmIvtfFKY

**Comment by tresf** [comment]
> [ ] All: QMdiWindow windowing controls are incorrectly colored (preview)

I installed QtCreator for Windows to see how these are decorated natively, and there's quite a difference... 

**WindowsVista**

```cpp
QApplication::setStyle(QStyleFactory::create("WindowsVista"));
```

![image](https://cloud.githubusercontent.com/assets/6345473/13604279/00f111e8-e510-11e5-8102-083b6add2f8a.png)

**Fusion**

```cpp
QApplication::setStyle(QStyleFactory::create("Fusion"));
```

![image](https://cloud.githubusercontent.com/assets/6345473/13604449/af297cd2-e510-11e5-9529-596f8156aad0.png)

**Windows** (classic)

```cpp
QApplication::setStyle(QStyleFactory::create("Windows"));
```

![image](https://cloud.githubusercontent.com/assets/6345473/13604853/c7bb2dac-e512-11e5-8d65-c7f67045c2aa.png)


Available styles (not available on all platforms):

![image](https://cloud.githubusercontent.com/assets/6345473/13604980/6a8c0b46-e513-11e5-9f95-46053f4d70bd.png)

Perhaps some compat code like this before the `QApplication` constructor... 

```cpp
#ifdef LMMS_BUILD_WIN32
    QApplication::setStyle("WindowsVista") || QApplication::setStyle("WindowsXP") || QApplication::setStyle("Windows");
#elif LMMS_BUILD_APPLE
    QApplication::setStyle("Macintosh");
#else
    QApplication::setStyle("GTK") || QApplication::setStyle("Fusion");
#endif
```

**Comment by tresf** [comment]
> [ ] All: QMdiWindow windowing controls are incorrectly colored (preview)


I guess the important lesson learned here is, be careful what you ask for, you just might get it.... (5f83afc).

Safe to say, we're going to leave this QMdiWindow windowing controls as-is.  Please speak up if you object. ;)

**Ubuntu 12.04**
![image](https://cloud.githubusercontent.com/assets/6345473/13614306/d9fd8cd2-e53d-11e5-9e32-e05b277aaee6.png)


**Windows XP**
![image](https://cloud.githubusercontent.com/assets/6345473/13613805/95d318c6-e53b-11e5-8c7c-d6c791fc6154.png)

**Windows 10**
![image](https://cloud.githubusercontent.com/assets/6345473/13613850/cde90446-e53b-11e5-9632-4679108edc9f.png)

**Mac OS 10.11**
![image](https://cloud.githubusercontent.com/assets/6345473/13614036/967366cc-e53c-11e5-90e5-4a1d92128c03.png)

**Comment by IvanMaldonado** [comment]
What will happen to the themeable title bars?

**Comment by tresf** [comment]
> What will happen to the themeable title bars?

> > Safe to say, we're going to leave this QMdiWindow windowing controls as-is (Plastique + Fusion). Please speak up if you object. ;)

We're going to roll our own still. :+1: Native look won't work as illustrated. :)

**Comment by musikBear** [comment]
I have made some more win32xp tests on the qt5PR.
Test for memory leak:

Idle out of focus
time     size in TM
18:50 : 163.768
18:55 : 163.792
19:00 : 163.820
19:05 : 164.080
19:10 : 164.092
19:10 : 164.080
19:15 : 164.088
19:30 : 164.100
19:40 : 164.100
19:45 : 164.100
19:50 : 164.100

Playing in focus
19:50 : 14.090
19:55 : 14.332
20:00 : 14.141
20:05 : 14.244
20:10 : 14.244
20:15 : 14.248
20:20 : 14.248

I dont understand why 'out of focus' is significantly larger in ram usage, but thats transparent in respect to the memory-leak test.
Imo These result shows that qt5 master has no memory-leaks. Good job, all!

Load and replay Test
I have gone through the whole library 'cool-songs'.
Every one loads extremely fast (few secd) and reliable. Every one replay perfectly. Imo relevance to
#2656 
But also to #1273

Test of ASIO4ALL
I tried a 3. party dll with ASIO4ALL. 
I could load and play, but not close, successful. Closing resulted in expected BSOD, and i did not retry.

**Comment by Umcaruje** [comment]
Here's a Qt5 specific crash: #2667

Also, the about window needs some love: 
![screenshot from 2016-03-11 15 23 19](https://cloud.githubusercontent.com/assets/6282045/13704648/3745f6f2-e79d-11e5-9f76-b3f6118079a0.png)

**Comment by tresf** [comment]
@Umcaruje, added to list, much obliged.

**Comment by musikBear** [comment]
some odd graphics *can* appear
There is no single action that reproducely will cause this, except mowing a lot of floating windows, and wait for it to happen.
![qt5graphicerr01](https://cloud.githubusercontent.com/assets/6662066/13723272/b6a7aa40-e85f-11e5-8287-3a26bde5a5ae.png)

This one here
![qt5graphicerr02](https://cloud.githubusercontent.com/assets/6662066/13723274/c5cc9dfa-e85f-11e5-8703-c9b51d015c23.png)
appeared after several track-containers had been moved up and down. Track-names has been almost halved vertically, and shuffled   
Obs! this could be my *screen*, an old tube-screen, actually from a dumpster :dollar: 
If noone else reports it too, it most likeli *is* my screen.

imo :
not sure this tearing its a big issue here in lmms. Any movement of the object will force a redraw, and everything is back to normal.

**Comment by StakeoutPunch** [comment]
@musikBear How much ram does your PC have, and what graphics (since it is a XP machine, probably older intel Q series, like Q945)?

I ask because it might be that these artifacts are from the age/specs of your machine. Older computers with very low integrated graphics (relative to today's standards) with low graphical memory or system ram tend to have random render tearing like you've shown.

**Comment by musikBear** [comment]
@StakeoutPunch True, its from 2007
Specs:
Processor Information:
    Vendor:  GenuineIntel
    Speed: 2671 Mhz
    2 logical processors
    2 physical processors
    HyperThreading:  Unsupported
    FCMOV:  Supported
    SSE2:  Supported
    
Network Information:
Network Speed:  DSL > 256k

Windows Version:
Windows XP (32 bit)
Network Information:
    Network Speed:  DSL > 256k

Windows Version:
    Windows XP (32 bit)
    NTFS:  Supported
    
Video Card:
    Driver:  NVIDIA GeForce 8600 GTS
    DirectX Driver Name:  nv4_disp.dll
    Driver Version:  6.14.11.5819
    DirectX Driver Version:  6.14.11.5819
    Driver Date: 12 Apr 2007
    Desktop Color Depth: 32 bits per pixel
    Monitor Refresh Rate: 85 Hz
    DirectX Card: NVIDIA GeForce 8600 GTS
    VendorID:  0x10de
    DeviceID:  0x400
    Number of Monitors:  1
    Number of Logical Video Cards:  1
    No SLI or Crossfire Detected
    Primary Display Resolution:  1024 x 768
    Desktop Resolution: 1024 x 768
    Primary Display Size: 12.60" x 9.45"  (15.75" diag)
                                            32.0cm x 24.0cm  (40.0cm diag)
                                            
Primary Bus: PCI Express 16x
    Primary VRAM: 256 MB
    Supported MSAA Modes:  2x 4x 8x 
    
Sound card:
    Audio device: Realtek HD Audio output
    Memory:
    RAM:  2047 Mb
    
Miscellaneous:
    UI Language:  English
    Microphone:  Yes, I have a microphone.
    Media Type:  DVD
    Total Hard Disk Space Available:  131061 Mb
    Largest Free Hard Disk Block:  12226 Mb

Mind you, i have successfully run 3 instances of lmms simultaneous on *that* hw!
I think it is one of lmms' strength, that it is able to run on even older hw, and stable on top of that. 
Dont pay attention to this. Screen-tearing is (imo) an absolutely unimportant issue. And it does only occur in real stress situations, not in normal use. Imo only the criticals are to be adressed, but thats only mho. B.r.

**Comment by Umcaruje** [comment]
Ok, so we can close this one out now, as all the issues were adressed. Great job everybody :+1:

**Comment by liushuyu** [comment]
:+1:

**Comment by tresf** [comment]
Although the checklist would suggest that, I feel we should make our build environment do this by default to consider this task closed.

**Comment by Umcaruje** [comment]
>I feel we should make our build environment do this by default to consider this task closed.

Hmm, you're right. Forgot about that one.

**Comment by tresf** [comment]
Added new item to the checklist: 
 [ ] Windows: Fix high-CPU usage for collapsed toolbars (#2767)

Goes without saying but this bug is a deal breaker on Windows if not resolved.  It will spike a 100% usage on a CPU core if running Qt5.  Discussion specific to the bug should remain in #2767.

**Comment by tresf** [comment]
Added new items to the checklist:

  * [ ] All: VSTs GUI does not load properly (#2855)

Yet another deal breaker for Qt5.  Help/workarounds, etc are appreciated. :+1:

**Comment by tresf** [comment]
Since [Windows VSTs seem to be in usable shape](https://youtu.be/exQwD1Kjsmg) and Mac does not yet support VSTs, I feel we are ready to switch to Qt5 for our next release.

I'm marking off the VST bug and adding a new task, "Remove Mac/Windows Qt4 builds from Travis-CI".

**Comment by simonvanderveldt** [comment]
Will building with Qt4 as an option be removed?

**Comment by tresf** [comment]
> Will building with Qt4 as an option be removed?

No, not right away as VST GUIs are still horribly broken with Qt5 on Linux. #2855.

We do have plans to shift Windows and Mac builds to Qt5 since they have Qt5 bundled --there is no downside.  Note, we may eventually offer Qt5 bundles for Linux as well per #2932, assuming VSTs are sorted.

**Comment by simonvanderveldt** [comment]
:thumbsup:

**Comment by tresf** [comment]
New issue... Qt5 prevents Mac ⌘+Drag copy as well as Mac ⌘+Drag Automation editor.

Although Copy/Paste has a right-click alternative, no such shortcut exists for linking to an automation pattern.

This may be unrelated to Qt5 (could be a code change) but my instinct is that it's directly Qt5 related.

**Comment by musikBear** [comment]
That "Drive not ready" thing?
![drivenotready](https://user-images.githubusercontent.com/6662066/33569145-0ae3830c-d929-11e7-9663-191a5918b760.JPG)
-that appears when lmms is loaded. I *suspect* that it is an error caused by the absence of a disk in Drive-B, or iow. This popup will occur, if the user do not have a SSD in drive-B, *and* the compiling is made on a system *with* a drive-B.
Could that be correct?

**Comment by tresf** [comment]
> Could that be correct?

Nope.

**Comment by tresf** [comment]
Closed via #4041.  The last remaining item was switching Travis to Qt5 and that is completed.
